Enhance internal logging (#3900)
open-telemetry/opentelemetry-go: CHANGELOG.md

Reviewer: There should be a comment that info has changed to v4 and error to v8

- The `WithoutTimestamps` option to `go.opentelemetry.io/otel/exporters/stdout/stdoutmetric` to sets all timestamps to zero. (#3828)
- The new `Exemplar` type is added to `go.opentelemetry.io/otel/sdk/metric/metricdata`.
  Both the `DataPoint` and `HistogramDataPoint` types from that package have a new field of `Exemplars` containing the sampled exemplars for their timeseries. (#3849)

> - The internal logging introduces a warning level equal to `V(8)` (verbosity level). (#3900)

Reviewer: ```suggestion
- The internal logging introduces a warning level equal to `V(1)` (verbosity level). (#3900)
```

> - The New `Warn` function is added to `internal/global` package for internal logging. (#3900)

Reviewer 2: This is probably not needed, the `internal` package is not accessible to the users we intend as the audience of this document. We should mention that the logging levels of the project have changed.

Author: Would it be more appropriate if I changed it to something like: 

- The logging levels of the project are updated as below: (#3900) 
  | Name | Severity |
  | -- | -- |
  | Error | 0 |
  | Warn | 1 |
  | Info | 4 |
  | Debug | 8 |

Reviewer 3: ```suggestion
- The internal logging introduces a warning level equal to `V(8)` (verbosity level) . (#3900)
```

> - The internal logging introduces a warning level equal to `V(8)` (verbosity level) . (#3900)

Reviewer 3: ```suggestion
- The internal logging introduces a warning level equal to `V(8)` (verbosity level). (#3900)
```

There was a typo in my suggestion. My bad, sorry.

> - The internal logging changes the verbosity level of info to `V(4)`, the verbosity level of debug to `V(8)`, and introduces a warning level equal to `V(1)`. (#3900)

Reviewer 3: I suggest moving it to `Changed` section (instead of `Added`)

Author: yeap agreed, actually, I've split it into two parts now, one line for _changed_ one line for _added_ section
